### PR TITLE
Use taskset 1 when checking for reproducible builds

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -85,6 +81,7 @@ test_linux_task:
 
   check_reproducible_script: |
     if [ -n "$CHECK_REPRODUCIBLE" ]; then
+      taskset 1
       make check_reproducible
```
Comment (ericmj) [inline .cirrus.yml:85]: Do we run things after `make check_reproducible`? If so should we reset the affinity after this command?

Comment (josevalim) [reply .cirrus.yml:85]: Reproducible is always the last one.

File: .cirrus.yml [hunk 2]
```diff
@@ -161,6 +158,7 @@ test_freebsd_task:
 
   check_reproducible_script: |
     if [ -n "$CHECK_REPRODUCIBLE" ]; then
+      taskset 1
```
Comment (bmwiedemann) [inline .cirrus.yml:161]: You need to write `taskset 1` that as the prefix of the actual command, so `taskset 1 gmake check_reproducible` - but if that does 2 builds, you only want to apply it to 1 of them.

Comment (josevalim) [reply .cirrus.yml]: Yup, thanks! The error message was very clear too. Unfortunately no taskset on macos, so I am shooting on the dark in the CI. :)

Comment (josevalim) [reply .cirrus.yml]: Sweet! I was able to reproduce the failure you got (not the same, but a similar one) with `taskset 1`. Now I brought the fix from master, let's see how it goes!

File: .cirrus.yml [hunk 3]
```diff
@@ -161,7 +157,7 @@ test_freebsd_task:
 
   check_reproducible_script: |
     if [ -n "$CHECK_REPRODUCIBLE" ]; then
-      gmake check_reproducible
+      taskset 1 gmake check_reproducible
```
Comment (josevalim) [inline .cirrus.yml:160]:
```suggestion
      gmake check_reproducible
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**